Content.Shared/Utility/EntityWhitelist.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System.Collections.Generic;
+using Content.Shared.GameObjects.Components.Tag;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Log;
+using Robust.Shared.Serialization;
+using Robust.Shared.Serialization.Manager.Attributes;
+
+namespace Content.Shared.Utility
+{
+    /// <summary>
+    ///     Used to determine whether an entity fits a certain whitelist
+    /// </summary>
+    /// <code>
+    /// whitelist:
+    ///   prototypes:
+    ///     - FireExtinguisher
+    ///     - DisgustingSweptSoup
+    ///   tags:
+    ///     - Cigarette
+    ///     - FirelockElectronics
+    ///   components:
+    ///     - Buckle
+    ///     - AsteroidRock
+    /// </code>
+    [DataDefinition]
+    public class EntityWhitelist : ISerializationHooks
+    {
+        /// <summary>
+        ///     Prototype IDs that are allowed in the whitelist.
+        /// </summary>
+        [DataField("prototypes")] public string[]? Prototypes = null;
```

Comment:
Relying on specific prototypes for behaviour is a no-no.

Reply:
i mean, it's generally not going to be used but i can't imagine that there are literally no use cases for this

Reply:
It goes against ECS, it's the reason we even made TagComponent.